plugins/org.eclipse.fordiac.ide.model/src/org/eclipse/fordiac/ide/model/dataimport/ConnectionHelper.java

```diff
@@ -10,39 +10,71 @@
 import java.util.EnumSet;
 import java.util.Set;
 
+import javax.xml.stream.XMLStreamException;
+
+import org.eclipse.emf.ecore.EClass;
+import org.eclipse.fordiac.ide.model.LibraryElementTags;
 import org.eclipse.fordiac.ide.model.Messages;
+import org.eclipse.fordiac.ide.model.NameRepository;
+import org.eclipse.fordiac.ide.model.data.DataType;
+import org.eclipse.fordiac.ide.model.dataimport.exceptions.TypeImportException;
+import org.eclipse.fordiac.ide.model.datatype.helper.IecTypes;
+import org.eclipse.fordiac.ide.model.errormarker.FordiacErrorMarkerInterfaceHelper;
+import org.eclipse.fordiac.ide.model.errormarker.FordiacMarkerHelper;
+import org.eclipse.fordiac.ide.model.libraryElement.AdapterConnection;
+import org.eclipse.fordiac.ide.model.libraryElement.Connection;
+import org.eclipse.fordiac.ide.model.libraryElement.ConnectionRoutingData;
+import org.eclipse.fordiac.ide.model.libraryElement.DataConnection;
+import org.eclipse.fordiac.ide.model.libraryElement.ErrorMarkerInterface;
+import org.eclipse.fordiac.ide.model.libraryElement.Event;
+import org.eclipse.fordiac.ide.model.libraryElement.EventConnection;
+import org.eclipse.fordiac.ide.model.libraryElement.FBNetworkElement;
 import org.eclipse.fordiac.ide.model.libraryElement.IInterfaceElement;
 import org.eclipse.fordiac.ide.model.libraryElement.InterfaceList;
+import org.eclipse.fordiac.ide.model.libraryElement.LibraryElementFactory;
+import org.eclipse.fordiac.ide.model.resource.TypeImportDiagnostic;
+import org.eclipse.fordiac.ide.model.typelibrary.AdapterTypeEntry;
+import org.eclipse.fordiac.ide.model.typelibrary.EventTypeLibrary;
 import org.eclipse.fordiac.ide.model.validation.LinkConstraints;
 
 public final class ConnectionHelper {
 
-	public static class ConnectionBuilder {
+	public static class ConnectionBuilder<T extends Connection> {
+
+		private final Set<ConnectionState> connectionState;
 
-		private Set<ConnectionState> connectionState;
+		final T connection;
+		final FBNetworkImporter importer;
 
-		private String destination;
+		private final String destinationString;
 		private InterfaceList destInterfaceList;
-		private IInterfaceElement destinationEndpoint;
 
-		private String source;
+		private final String sourceString;
 		private InterfaceList srcInterfaceList;
-		private IInterfaceElement sourceEndpoint;
 
-		public ConnectionBuilder(final String source, final String destination) {
+		public ConnectionBuilder(final String sourceString, final String destinationString, final T connection,
+				final FBNetworkImporter importer) {
 			this.connectionState = EnumSet.of(ConnectionState.VALID);
-			this.source = source;
-			this.destination = destination;
+			this.sourceString = sourceString;
+			this.destinationString = destinationString;
+			this.connection = connection;
+			this.importer = importer;
+
+			connection.setSource(getConnectionEndPoint(sourceString, false));
+			connection.setDestination(getConnectionEndPoint(destinationString, true));
```

Comment:
How about we always resolve and validate connections at the end, also for regular networks? That should save us the extra repair step and also make regular networks immune to the order of XML elements?

Reply:
with the work I did here we could reconsider later. But as the order in the xml is defined by the dtd I see less need for all networks.